OpenRA.Game/FileFormats/WavLoader.cs

```diff
@@ -33,32 +33,33 @@ public class WavLoader : ISoundLoader
 		public enum WaveType { Pcm = 0x1, ImaAdpcm = 0x11 }
 		public static WaveType Type { get; private set; }
 
+		bool wavDetected;
+
 		bool LoadSound(Stream s)
 		{
-			var type = s.ReadASCII(4);
```

Comment:
Nope. In this case I am ahead already. You can test this yourself by reordering to `Aud, Wav` at https://github.com/OpenRA/ra2/pull/28/files#diff-1ea1b12ca5110238b7484e0a950b8f62R234 to make sure detection works in both cases. The `bool wavDetected` stays false if it isn't a WAVE file although I don't stop at byte 4, but force it to parse the whole stream.